torch/_inductor/codegen/common.py

```diff
@@ -222,6 +222,18 @@ def init_backend_registration():
     if get_scheduling_for_device("xpu") is None:
         register_backend_for_device("xpu", TritonScheduling, WrapperCodeGen)
 
+    private_backend = torch._C._get_privateuse1_backend_name()
+    if private_backend != "privateuseone":
+        if get_scheduling_for_device(private_backend) is None:
+            from torch.utils.backend_registration import _get_custom_mod_func
+            try:
+                triton_scheduling = _get_custom_mod_func("TritonScheduling")
+                wrapper_codegen = _get_custom_mod_func("WrapperCodeGen")
+                if triton_scheduling and wrapper_codegen:
+                    register_backend_for_device(private_backend, triton_scheduling, wrapper_codegen)
```

Comment:
You can call this function from outside PyTorch tree as well.

Reply:
@jansel Thanks for your comments.
When calling this function out-of-tree, some profiler testcases will fail.
You can reproduce it using the following method:
Add below codes at： https://github.com/pytorch/pytorch/blob/main/test/profiler/test_profiler.py#L347
```
        from torch._inductor.codegen.common import register_backend_for_device                                                                                                                                    
        from torch._inductor.codegen.triton import TritonScheduling
        from torch._inductor.codegen.wrapper import WrapperCodeGen
             
        register_backend_for_device("apu", TritonScheduling, WrapperCodeGen)
```
and then execute `python test/profiler/test_profiler.py -k TestProfiler.test_source_multithreaded_basic_work_in_main_thread_False`.
The following error will be encountered:
```
======================================================================
FAIL: test_source_multithreaded_basic_work_in_main_thread_False (__main__.TestProfiler)
Test various threading configurations.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/zhouyusong/venv_2.4/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2739, in wrapper
    method(*args, **kwargs)
  File "/home/zhouyusong/venv_2.4/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 446, in instantiated_test
    test(self, **param_kwargs)
  File "/home/zhouyusong/pytorch_2.4/test/profiler/test_profiler.py", line 459, in test_source_multithreaded
    self.assertEqual(
  File "/home/zhouyusong/venv_2.4/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3631, in assertEqual
    raise error_metas.pop()[0].to_error(
AssertionError: Scalars are not equal!

Expected 2 but got 3.
Absolute difference: 1
Relative difference: 0.5
2, Counter({2: 96, 1: 63, 18446744073709551615: 5})
```
It appears that using the out-of-tree method prematurely triggers the import of some Python modules.
I am still trying to locate and resolve the issue.